Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftWindowsEvents.yaml

```diff
@@ -50,122 +50,209 @@ ParserParams:
+    // Active Directory Replica Source Naming Context Event Ids
+    let ActiveDirectoryReplicaIds = dynamic([4929]);
+    // Firewall Event Ids
+    let FirewallEventIds = dynamic([5025,5027,5028,5029,5030,5034,5035,5037]);
+    // EventID Lookup
+    let EventIDLookup = datatable(EventID:int, Operation:string, EventType:string, Object:string, ObjectType:string)
+    [   
+        1102, "Delete Logs", "Delete", "Security Logs", "Event Log",
+        4698, "Create Scheduled Task", "Create", "", "Scheduled Task",
+        4699, "Delete Scheduled Task", "Delete", "", "Scheduled Task",
+        4700, "Enable Scheduled Task", "Enable", "", "Scheduled Task",
+        4701, "Disable Scheduled Task ", "Disable", "", "Scheduled Task",
+        4702, "Update Scheduled Task", "Set", "", "Scheduled Task",
+        4929, "Remove Active Directory Replica Source Naming Context", "Delete", "", "Other",
+        5025, "Stop Firewall Service", "Disable", "Firewall Service", "Service",
+        5027, "Retrieve the Security Policy From The Local Storage", "Read", "Firewall Service", "Service",
+        5028, "Parse the new Security Policy", "Set", "Firewall Service", "Service",
+        5029, "Initialize the Firewall Driver", "Initialize", "Firewall Service", "Service",
+        5030, "Start the Firewall Service", "Start", "Firewall Service", "Service",
+        5034, "Stop Firewall Driver", "Stop", "Firewall Driver", "Driver",
+        5035, "Start Firewall Driver", "Start", "Firewall Driver", "Driver",
+        5037, "Terminating Firewall Driver", "Terminate", "Firewall Driver", "Driver"
+    ];
+    let FilteredEventIds = toscalar(EventIDLookup 
+        | where (array_length(eventtype_in) == 0 or EventType in (eventtype_in))
+            and (array_length(operation_has_any) == 0 or Operation has_any (operation_has_any))
+        | summarize make_set(EventID)
+    );
+    let FilteredParsedEventIds = iif(array_length(FilteredEventIds) >0, FilteredEventIds, ParsedEventIds);
+    let ParsedEvents =
+    union (
+        union (
+        // SecurityEvents
             SecurityEvent
-            | where 
-                (isnull(starttime) or TimeGenerated >= starttime) 
+            | where not(disabled)
+                and (isnull(starttime) or TimeGenerated >= starttime) 
                 and (isnull(endtime) or TimeGenerated <= endtime)
-                and not(disabled)
-            | project EventID, EventData, _ResourceId, TimeGenerated, Computer, Type
-            | where EventID == 1102
+                and (array_length(srcipaddr_has_any_prefix) == 0)
+                and (array_length(actorusername_has_any) == 0 or EventData has_any (actorusername_has_any))
+                and (array_length(newvalue_has_any) == 0 or EventData has_any (newvalue_has_any))
+                and EventID in(FilteredParsedEventIds)
+                and (array_length(object_has_any) == 0 or EventData has_any(object_has_any))           
```

Comment:
This filter will not work as expected. For example, for Object == "Security Logs", the text will not be part of the event and thus would get dropped even when it is required.

Reply:
Done